next.clayui.com/content/docs/foundations/composing.md

```diff
@@ -3,4 +3,146 @@ title: "Composing"
 description: "Component composition philosophy"
 ---
 
-This is a stub. Help our community expand it.
\ No newline at end of file
+<div class="nav-toc-absolute">
+<div class="nav-toc">
+
+- [Composition](#composition)
+- [Philosophy](#philosophy)
+    - [Low-level components](#low-level-components)
+	- [High-level components](#high-level-components)
+
+</div>
+</div>
+
+Clay follows some fundamentals for creating new components and as we distribute the components so that they can be used, the components follow some of the main React patterns: [composition](https://reactjs.org/docs/composition-vs-inheritance.html) or you can listen around for "bulding blocks".
+
+We believe that using composition is the most correct and is the most widely used standard in the community. We want to allow people to reuse our components for different purposes, we understand that we have different people with different opinions on how we want to use Clay for their purposes. We think about how we can have a little of both worlds.
+
+## Composition
+
+The composition is very strong with React, JSX syntactic sugar makes it very powerful and more viable when coupled with special `children` prop for passing arbitrary child elements. Props in React components allows us to pass objects and consequently React elements since they are objects.
+
+```jsx
+<ClayModal>
+	<ClayModal.Footer
+		first={
+			<ClayButton.Group spaced>
+				<ClayButton displayType="secondary">
+					{'Secondary'}
+				</ClayButton>
+				<ClayButton displayType="secondary">
+					{'Secondary'}
+				</ClayButton>
+			</ClayButton.Group>
+		}
+		middle={
+			<ClayButton>{'Primary'}</ClayButton>
+		}
+	/>
+</ClayModal>
+```
+
+So we use some of the composable component approach in React.
```

Comment:
Not really sure what this sentence it trying to explain. Is this a title for the content below?

Reply:
Yeah, just one sentence to get into composing methods. Perhaps it would be more explicit:

```suggestion
So we follow some of React's composition approaches:
```